src/js/lib/monster.socket.js

```diff
@@ -90,45 +114,48 @@ define(function(require) {
 			};
 		},
 
-		// Reconnect loop that retries to connect after {{startingTimeout}} ms
-		// In case it fails, we multiply the previous timeout by {{multiplier}}, and then try again once that time has gone by
-		// The max timer is set to {{maxtimeout}}
-		connect: function() {
+		/**
+		 * Reconnect loop that retries to connect after {{startingTimeout}} ms
+		 * In case it fails, we multiply the previous timeout by {{multiplier}}, and then try again once that time has gone by
+		 * The max timer is set to {{maxTimeout}}
+		 * @param  {Number} [pTimeoutDuration=0]
+		 */
+		connect: function(pTimeoutDuration) {
 			var self = this,
-				timeoutDuration,
-				delayBeforeConnect = 0,
+				timeoutDuration = _.isUndefined(pTimeoutDuration) ? 0 : pTimeoutDuration,
 				startingTimeout = 250,
-				maxTimeout = 1000 * 60,
 				multiplier = 2,
-				connectAttempt = function() {
-					if (!self.connected) {
-						self.initialize();
-
-						if (timeoutDuration) {
-							if ((timeoutDuration * multiplier) < maxTimeout) {
-								timeoutDuration *= multiplier;
-							} else {
-								timeoutDuration = maxTimeout;
-							}
+				maxTimeout = 60 * 1000;
 
-							self.log('Attempting to reconnect to WebSocket at ' + monster.util.toFriendlyDate(new Date()));
-							self.log('Next try in ' + timeoutDuration / 1000 + ' seconds');
-						} else {
-							timeoutDuration = startingTimeout;
+			if (!monster.util.isLoggedIn()) {
+				return self.log('Unable to connect to WebSocket while logged out');
+			}
+			if (self.isConnecting) {
+				return self.log('Already attempting to connect to WebSocket');
```

Comment:
If the socket connection takes too long to fail, no more attempts will be done if this get executed in the meantime on a second attempt. Is this the expected behavior for the time being?

Reply:
It'd be nice not to introduce regressions, let me add an error callback to handle that.